core/modules/memman/MemMan.cc

```diff
@@ -0,0 +1,43 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2015 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
```

Comment:
I think the intent behind making an implementation file Foo.cc include its header Foo.h first no matter what is that the compiler will then verify that Foo.h has included everything needed to make itself useable by others. Otherwise it's easy to forget to include say `<string>` in Foo.h because Foo.cc includes it before including Foo.h. At that point someone else using Foo.h but not `<string>` will run into compile errors.